### DISCUSSION TITLE
[@types/yargs] command/usage returned type doesn't include the commands defined arguments

### DISCUSSION THREAD

**Discussion by segevfiner** [DefinitelyTyped/DefinitelyTyped#67512] [Issues with a @types package]
The returned object from `yargs` parsing include the arguments of the sub-command that was ran. Sometimes instead of using `handler`, you want to parse the arguments of the main program only, or handle commands in some other way:
```ts
const argv = yargs(hideBin(process.argv))
  .command("$0 <file>", "", (yargs) => {
    return yargs.positional("file", { type: "string" });
  }, (argv) => {
    console.log(argv);
  })
  .strict()
  .parseSync();
console.log(argv);
```

Inside the handler, you have types for the `file` argument, but outside of it, you don't. Despite both `console.log` printing the same.

Also, it seems you need to return the `yargs` from the `builder` argument of the command for types to work, which is not documented in the official yargs docs, and you might want to document this more clearly in the TSDoc of the `command` and `usage` methods.

[a comment by an automated account is not shown]

**Comment by segevfiner** [comment]
Ping? @poelstra, @mizunashi-mana, @pushplay, @JimiC, @steffenvv, @forivall, @ExE-Boss, @Aankhen, @bcoe

**Reply by forivall** [reply]
Feel free to open a PR! Let us know if you need help.

**Reply by segevfiner** [reply]
This seems like a deliberate change, but still kinda undesirable for a script that doesn't have subcommands... See https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/62150 for further discussion.
